packages/@okta/vuepress-site/docs/reference/api/apps/index.md

```diff
@@ -5133,6 +5133,981 @@ curl -v -X GET \
 }
 ```
 
+## Application client secret management operations
+
+### Add new client secret
+
+<ApiOperation method="post" url="/api/v1/apps/${applicationId}/credentials/secrets" />
+
+Adds a new secret to the client's collection of secrets
+
+> **Note:** This API lets you bring your own secret. If no secrets are specified in the request, it adds a new system-generated secret.
+
+##### Request parameters
+
+| Parameter     | Description                                                                     | Param Type | DataType                                      | Required | Default |
+| ------------- | ------------------------------------------------------------------------------- | ---------- | --------------------------------------------- | -------- | ------- |
+| applicationId | Unique key of an [Application](#application-object)                                 | URL        | String                                    | TRUE     |         |
+| metadata | Metadata for creating a secret [Client Secret Metadata](#client-secret-request-object)   | Body      | Client Secret Metadata                     | FALSE    |         |
+
+##### Response parameters
+
+Returns the generated [Client Secret](#client-secret-response-object)
+
+##### Request example
+
+```bash
+curl -v -X POST \
+-H "Accept: application/pkcs10" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+-d '{
+    "client_secret" : "3vimrC5Yv6bSDJzrUdLEYvkf9ElwUeWdndO5nhYp"
+}' "https://${yourOktaDomain}/api/v1/apps/0oad5lTSBOMUBOBVVQSC/credentials/secrets/"
+```
+
+##### Response example
+
+```json
+{
+  "id": "ocsap6nz61rKdsoyOY0h7",
+  "client_secret": "3vimrC5Yv6bSDJzrUdLEYvkf9ElwUeWdndO5nhYp",
+  "secret_hash": "0xbeefbeefbeefbeefbeefbeef",
+  "created": "2022-01-01T04:22:27.000Z",
+  "status": "ACTIVE",
+  "_links" : {
+    "deactivate": {
+      "href": "https://${yourOktaDomain}/api/v1/apps/0oad5lTSBOMUBOBVVQSC/credentials/secrets/ocsap6nz61rKdsoyOY0h7/lifecycle/deactivate",
+      "hints": {
+        "allow": [
+          "POST"
+        ]
+      }
+    }
+  }
+}
+```
+
+If `client_secret` in the request is more than 100 characters long, you receive an error response.
+
+```json
+{
+  "errorCode": "E0000001",
+  "errorSummary": "Api validation failed: client_secret",
+  "errorLink": "E0000001",
+  "errorId": "oaeMHrsk2WLTACvPU5T7yQ4yw",
+  "errorCauses": [
+    {
+      "errorSummary": "client_secret: 'client_secret' cannot be more than '100' characters long."
+    }
+  ]
+}
+```
+
+If `client_secret` in the request is less than 14 characters long, you receive an error response.
+
+```json
+{
+  "errorCode": "E0000001",
+  "errorSummary": "Api validation failed: client_secret",
+  "errorLink": "E0000001",
+  "errorId": "oaeMHrsk2WLTACvPU5T7yQ4yw",
+  "errorCauses": [
+    {
+      "errorSummary": "client_secret: The field is too short"
```

Comment:
Wouldn't it make more sense to have the opposite error message as is listed on line 5201? I don't see why these should be significantly different as it just makes it less consistent.

Reply:
This error message comes from the OAuth2ClientSecretMediated Validator and the other 2 are from separate validation we have. I will need to make changes in the API first to make the change here.

Reply:
Can we create a follow-up ticket for this? You don't need to fix it now, but it's a matter of developer experience to have consistent error messages being returned.

Reply:
https://oktainc.atlassian.net/browse/OKTA-462333

Reply:
updated